packages/react-native-renderer/src/NativeMethodsMixin.js

```diff
@@ -204,6 +206,12 @@ if (__DEV__) {
   NativeMethodsMixin_DEV.UNSAFE_componentWillReceiveProps = function(newProps) {
     throwOnStylesProp(this, newProps);
   };
+
+  // React may warn about cWM/cWRP/cWU methods being deprecated.
+  // Add a flag to suppress these warnings for this special case.
+  // TODO (bvaughn) Remove this flag once the above methods have been removed.
+  NativeMethodsMixin_DEV.componentWillMount.__suppressDeprecationWarning = true;
+  NativeMethodsMixin_DEV.componentWillReceiveProps.__suppressDeprecationWarning = true;
```

Comment:
Do any first- or third-party components use this with `createReactClass` like
```js
createClass({
  mixins: [NativeMethodsMixin],
```

?

If yes, what happens if such component declares `componentWillReceiveProps`, but not `componentWillMount`? Could it be that suppression [here](https://github.com/facebook/react/blob/f05296baf586dc8e6d6e6a3e6d666c8dd8e4ccd5/packages/react-reconciler/src/ReactStrictModeWarnings.js#L208) would also show no message for custom `componentWillReceiveProps` (which we *should* have warned about)?

Reply:
Many built-in components like `View` and `ActivityIndicator` use `createReactClass` with `NativeMethodsMixin` in this way. None of them use legacy `componentWillReceiveProps` or `componentWillMount` methods though because I've code-modded them all away (before enabling this warning for the RN renderer).

I don't know if third-party components use the mixin, but hopefully not since it's behind `__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED`.

Reply:
Meh. Probably okay.

Reply:
You are correct though, that if I was wrong about the 2nd point, and a component only declared cWRP, the warning would be suppressed.

I _could_ expand the check you linked to to verify that flag on _both_ of the lifecycles if you think it's worth doing. Lemme see how involved this would be. Probably not too bad.

Reply:
No worries. It was trivial to make the check robust enough to cover that case too.

Reply:
FWIW, the [potentially troublesome check](https://github.com/facebook/react/blob/f05296baf586dc8e6d6e6a3e6d666c8dd8e4ccd5/packages/react-reconciler/src/ReactStrictModeWarnings.js#L208) was later improved via facebook/react/pull/12647